### DISCUSSION TITLE
Release of finetuning evaluation code

### DISCUSSION THREAD

**Discussion by ricklisz** [MIC-DKFZ/ssl3d#3] [General]
Dear SSL3D challenge organizers,

Hi, I wonder when will the code and datasets for finetuning evaluation will be released ? 

Best
Rick

**Comment by constantinulrich** [comment]
Hi, we will not provide the fine-tuning data. Participants will only submit their pretrained weights, and we will do the fine-tuning on probably 6 hidden datasets. However, the following information on how we do the fine-tuning will be added to the website in the next days: 

Best, 
Constantin


## Downstream Fine-Tuning
We will provide some information about the "hidden" downstream tasks that may have an impact on your pre-training.
The downstream datasets can have multiple input channels. We repeat the input stem weights to fit the number of input channels of the downstream task.


Sources for public data where you could test your pre-training:
- [Isles2022](https://zenodo.org/records/6362388#.YnuTnxNBzmF) 
- [Brats2023](https://www.synapse.org/Synapse:syn53708249) 
- [Yale brain metastases](https://www.cancerimagingarchive.net/collection/pretreat-metstobrain-masks/)

# Segmentation
- z-score normalization
- cubic 1mm target spacing
- [160,160,160] patch size
We use this [repository](https://github.com/TaWald/nnUNet) for segmentation fine-tuning. We will use the trainer "PretrainedTrainer_150ep" and "PretrainedTrainer_Primus_150ep"!

# Classification
- z-score normalization
- cubic 1mm target spacing
- Apply [HD-bet](https://github.com/MIC-DKFZ/HD-BET) to find the center of the brain
- Then, we use a [160,160,160] patch around this center for training
We use [this](https://github.com/constantinulrich/SSL3D_classification) repository for classification fine-tuning. The exact hyperparameters will vary between the datasets, but similar to the example yaml file in the readme.

**Reply by Tenbatsu24** [reply]
Hello,

Could you please confirm whether the training will follow a 50-50 split for train and validation in the segmentation task, and use 5-fold cross-validation for downstream classification? According to the [preprint](https://arxiv.org/pdf/2412.17041), this setup is described in Section 3.1 and Appendix C.2.

If a different methodology is planned for fine-tuning, may I kindly ask you to share the details?

Thank you for organizing this challenge!

**Comment by constantinulrich** [comment]
Hi, we are doing a 3-fold cross-validation. That was the best trade-off between compute resources and "stable" results. 

See also [here](https://ssl3d-challenge.dkfz.de/downstream_finetuning ) for a slightly updated preprocessing pipeline and explanation.
Best,
Constantin

**Reply by Tenbatsu24** [reply]
Hello,

 Thank you for your previous response. I have a follow-up question regarding the evaluation metrics. For the downstream fine-tuning, are the reported average metrics based on the best validation checkpoints (checkpoint_best) from each fold, or from the final epoch checkpoints (checkpoint_last)? This wasn’t entirely clear in the classification and segmentation fine-tuning sections. In particular, I’ve noticed a large difference between the two strategies for classification where some baseline methods suffer from overfitting but other did not.

Thank you for your time!

**Reply by constantinulrich** [reply]
Hi, the 'best_checkpoint" from nnunet is determined the pseudo dice on random 50 foreground patches. Its not actually a true best checkpoint. We always use the last checkpoint.

**Reply by constantinulrich** [reply]
For classification, we selected the hyperparameter in such way thad we did not observe severe overfitting for the testing datasets. But in general, the classification results are more affected by noise compared to segmentation
